### DISCUSSION TITLE
[Bug report] Git repository Test Connection doesn't return success or failure

### DISCUSSION THREAD

**Discussion by sajeshcherian** [GetPublii/Publii#1420] [Bug Report]
### Operating system

Pop!_OS 22.04 LTS

### Publii version

Version: 0.45.1 (build 16608)

### Issue type

Application

### Bug description

Test Connection doesn't return success or failure and keeps trying to connect even if the connection is a success. I am able to post to my blog via Git repository despite that.

### Steps to reproduce

1. Set server type to Git repository.
2. Fill in required fields.
3. Test Connection.
4. The button says Checking Connection and never responds with success or failure.
5. Able to publish on my blog with the same settings.

### Relevant log output

```shell
deployment-errors.log is empty.

deployment-process.log below
[[
...
[i] Git debug: git add done
[i] Git debug: changes exists =  true
[i] Git debug: commit done
[i] Git debug - message: . Processing 1 references

[i] Git debug - message: Processed 1 references in total

[i] Git debug: push done
]]

regenerate-errors.log
[[

(process:145571): GLib-CRITICAL **: 08:38:48.907: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:48.955: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.045: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.065: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.210: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.605: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.978: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:49.994: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:50.278: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:50.294: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:50.295: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:50.502: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:50.905: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:51.032: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:51.067: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

(process:145571): GLib-CRITICAL **: 08:38:51.250: g_datalist_id_set_data_full: assertion 'key_id > 0' failed

***MEMORY-ERROR***: [145571]: GSlice: assertion failed: sinfo->n_allocated > 0
]]
```

**Comment by example-user30** [comment]
I also observed this same functionality.  Be aware that you must use the [personal access token](https://www.geeksforgeeks.org/how-to-generate-personal-access-token-in-github/) instead of user/password in GitHub to successfully connect.  Even when this is done the test connection does not work.

**Comment by dziudek** [comment]
@example-user30 @sajeshcherian - this problem has been reported few times, but I am still unable to reproduce this on my local instance. I have even tried to add few modifications, but still something else is happening what prevents test connection from returning proper state :/

**Comment by jaansvrcek** [comment]
I am having the same issue:
Two different sites on different gitlab repos. 
The first one works like charm -> Test connection returns positive reply.
The other one though doesn't return anything when clicking "Test connection" -> the button got stuck on "Checking connection..."

The configuration of the pages is almost identical, the only change is the token for the appropriate repo.
When calling https://gitlab.com/api/v4/personal_access_tokens/self  using curl with used token in the header I do get a affirmation of the valid token.

Both of the logs: deployment-process.log and  deployment-errors.log  are empty.

Publii Version 0.45.2 (0.45.2) MacOS Sonoma 14.2.1 (23C71)
